bot/exts/moderation/infraction/infractions.py

```diff
@@ -406,7 +407,9 @@ async def apply_kick(self, ctx: Context, user: Member, reason: t.Optional[str],
         if reason:
             reason = textwrap.shorten(reason, width=512, placeholder="...")
 
-        action = user.kick(reason=reason)
+        async def action() -> None:
```

Comment:
I'd like to see comments here or elsewhere saying why we're not passing around coroutines, otherwise in the future someone might look at it and think this should be refactored.

Reply:
Added a note to the functions docstrings